File: src/atdd/coder/validators/fixtures/silent_swallow/python_violations/seek_opponent.py
"""Fixture: real-incident shape — silent swallow returning broken success value."""

import logging

logger = logging.getLogger(__name__)


def seek_opponent_silent_return(player_id: str) -> str:
    try:
        return _create_match(player_id).id
    except Exception:
        return ""


def seek_opponent_pass_then_fall_through(player_id: str) -> str:
    try:
        return _create_match(player_id).id
    except Exception:
        pass
    return ""


def seek_opponent_bare_except(player_id: str) -> str:
    try:
        return _create_match(player_id).id
    except:  # noqa: E722
        return "fallback"


def seek_opponent_typed_except(player_id: str) -> str:
    try:
        return _create_match(player_id).id
    except (ValueError, KeyError):
        return "default"


class _Match:
    id: str = ""


def _create_match(player_id: str) -> _Match:
    return _Match()

File: src/atdd/coder/validators/fixtures/silent_swallow/typescript_violations/seekOpponent.ts
// Fixture: TypeScript silent-swallow violations.

interface Match {
  id: string;
}

declare function createMatch(playerId: string): Match;
declare function parsePayload(p: string): unknown;
declare const defaultValue: unknown;

export function seekOpponentSilentReturn(playerId: string): string {
  try {
    return createMatch(playerId).id;
  } catch (e) {
    return "";
  }
}

export function seekOpponentEmptyCatch(playerId: string): string {
  try {
    return createMatch(playerId).id;
  } catch {
    return "fallback";
  }
}

export function parseWithFallback(payload: string): unknown {
  try {
    return parsePayload(payload);
  } catch (e) {
    return defaultValue;
  }
}
